### PR TITLE
MCP-2166 claim id type overhaul

### PR DESCRIPTION
**Pull request by ianwolson**
## What was the problem?
We need to find claims using claimSubmissionId and idType because claimSubmissionId by itself is not unique.

Associated tickets or Slack threads:

- [MCP-2166](https://amida.atlassian.net/browse/MCP-2166)

## How to test this PR
- Step 1
- Step 2[^secrel]

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: service/provider/src/main/java/gov/va/vro/service/provider/services/AssessmentResultProcessor.java [hunk 1]
```diff
@@ -19,8 +17,9 @@ public class AssessmentResultProcessor implements Processor {
 
   @Override
   public void process(Exchange exchange) {
-    UUID claimId = exchange.getProperty("claim-id", UUID.class);
-    if (claimId == null) {
+    String claimSubmissionId = exchange.getProperty("claimSubmissionId", String.class);
+    String idType = exchange.getProperty("idType", String.class);
+    if (claimSubmissionId == null) {
```
Comment (dimitri-amida) [inline service/provider/src/main/java/gov/va/vro/service/provider/services/AssessmentResultProcessor.java:22]: what if idType is null?

File: service/db/src/main/java/gov/va/vro/service/db/SaveToDbServiceImpl.java [hunk 2]
```diff
@@ -70,7 +73,9 @@ public void insertAssessmentResult(
   @Override
   public void insertEvidenceSummaryDocument(GeneratePdfPayload request, String documentName) {
     ClaimEntity claim =
-        claimRepository.findByClaimSubmissionId(request.getClaimSubmissionId()).orElse(null);
+        claimRepository
+            .findByClaimSubmissionIdAndIdType(request.getClaimSubmissionId(), request.getIdType())
+            .orElse(null);
```
Comment (dimitri-amida) [inline service/db/src/main/java/gov/va/vro/service/db/SaveToDbServiceImpl.java:78]: I don't see why convert to null here. You can just check if the optional is present

File: service/db/src/main/java/gov/va/vro/service/db/SaveToDbServiceImpl.java [hunk 3]
```diff
@@ -45,8 +44,12 @@ public Claim insertClaim(Claim claim) {
...
   @Override
   public void insertAssessmentResult(
-      UUID claimId, AbdEvidenceWithSummary evidenceResponse, String diagnosticCode) {
-    ClaimEntity claimEntity = claimRepository.findById(claimId).orElse(null);
+      String claimSubmissionId,
+      String idType,
+      AbdEvidenceWithSummary evidenceResponse,
+      String diagnosticCode) {
+    ClaimEntity claimEntity =
+        claimRepository.findByClaimSubmissionIdAndIdType(claimSubmissionId, idType).orElse(null);
```
Comment (dimitri-amida) [inline service/db/src/main/java/gov/va/vro/service/db/SaveToDbServiceImpl.java:52]: I don't see why convert to null here. You can just check if the optional is present

**Comment by dimitri-amida** [comment]
@ianwolson in ClaimRepository the method "Optional<ClaimEntity> findByClaimSubmissionId(String claimSubmissionId);" should either return a list or not exist at all, since we have established that the id alone is not sufficient to identify the claim.

**Review by dimitri-amida — APPROVED** [review]
added some comments

**Review by jbregstone — APPROVED** [review]

File: service/provider/src/main/java/gov/va/vro/service/provider/camel/PrimaryRoutes.java [hunk 4]
```diff
@@ -66,13 +66,15 @@ private void configureRouteClaimSubmitForFull() {
...
-        .setProperty("claim-id", simple("${body.recordId}"))
+        .setProperty("claimSubmissionId", simple("${body.claimSubmissionId}"))
+        .setProperty("idType", simple("${body.idType}"))
         .routingSlip(method(slipClaimSubmitRouter, "routeClaimSubmit"))
         .routingSlip(method(slipClaimSubmitRouter, "routeHealthAssess"))
         .process(assessmentResultProcessor);
   }
 
   private void configureRouteGeneratePdf() {
+    String test = "test";
```
Comment (jbregstone) [inline service/provider/src/main/java/gov/va/vro/service/provider/camel/PrimaryRoutes.java:77]: did you mean to leave this line here?

**Review by anthonyvelazquez — APPROVED** [review]

**Comment by au2** [comment]
@ianwolson is this supposed to go in with the changes @alan-amida and you are doing.

**Comment by au2** [comment]
@ianwolson is this supposed to go in after the changes you and @alan-amida are doing?

**Comment by ianwolson** [comment]
@au2 yes im doing the changes on that branch now as we're making changes to the claim table, we can close this branch

**Comment by au2** [comment]
Closing as instructed.

**Pull request closed without merging**